src/main/java/core/mate/academy/service/BulldozerProducer.java

```diff
@@ -0,0 +1,12 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Machine;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Machine> {

```

Comment:
```suggestion
public class BulldozerProducer implements MachineProducer<Bulldozer> {

```

Логичней параметризировать MachineProducer именно тем классом машин, которые он должен генерировать (то же для всех остальных MachineProducer)